### DISCUSSION TITLE
Custom keybinding to apply Completion Snippets

### DISCUSSION THREAD

**Discussion by artyom-podymov** [microsoft/monaco-editor#3650] [Q&A]
Hi all,
In [Completion provider example](https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-completion-provider-example) if I type `my-third-party-library` and resolve suggestion for it, the monaco editor inserts snippets.
![image](https://user-images.githubusercontent.com/28300948/140984353-a11e2b78-bb43-406a-9655-170ca8efc1d3.png)
I can navigate between them using `Tab` / `Shift+Tab`, and to close this mode I need to go through all of them (using `Tab`) or press `CtrlCmd+Enter`

**The question is, is it possible to close this mode by using some custom keybinding?**

For example, I want to use the `Enter` for this purpose, but at the moment a new line is inserted instead (I want to prevent such behavior).
![image](https://user-images.githubusercontent.com/28300948/140985686-2f7ae587-3ea2-4918-bfb3-18fc64f5a256.png)

**Comment by hediet** [comment]
I think this is possible, most contributions just register a keybinding for `esc` with some when clauses.

**Comment by tamayika** [comment]
Use this code.

```ts
monaco.editor.addKeybindingRule({
    keybinding: monaco.KeyCode.Enter,
    command: "jumpToNextSnippetPlaceholder",
    when: "editorTextFocus && hasNextTabstop && inSnippetMode"
})
```
